src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -690,7 +699,7 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
         {
             DeconstructionVariable locals = BindDeconstructionVariables(left, diagnostics, ref declaration, ref expression);
```

Comment:
>BindDeconstructionVariables [](start = 44, length = 27)

Is there still an advantage of using ```NestedVariables```? Could this function produce tuple literal right from the beginning instead of having to translate this structure later on?